lib/hydra/works/services/generic_file/add_file_to_generic_file.rb

```diff
@@ -8,7 +8,7 @@ class AddFileToGenericFile
     # @param [Boolean] update_existing whether to update an existing file if there is one. When set to true, performs a create_or_update. When set to false, always creates a new file within generic_file.files.
     # @param [Boolean] versioning whether to create new version entries (only applicable if +type+ corresponds to a versionable file)
 
-    def self.call(generic_file, file, type, update_existing: true, versioning: true)
+    def self.call(generic_file, file, type, update_existing: true, versioning: true, mime_type: nil)
```

Comment:
Just a note that the MIME type-related changes to this file are the ones @grosscol is working on in hydra-derivatives, so this is slated to go away once that's done.